scripts/buildWeeklyRequestCountDocuments/index.js

```diff
@@ -0,0 +1,53 @@
+const admin = require('firebase-admin')
+const { initFirebaseAdmin } = require('../utils/initFirebaseAdmin.js')
+const { firebaseTimestampToMoment } = require('../../mixins/date/firebaseTimestampToMoment')
+const { getFormattedEndOfWeek } = require('../../mixins/date/getFormattedEndOfWeek')
+const { getFormattedStartOfWeek } = require('../../mixins/date/getFormattedStartOfWeek')
+
+initFirebaseAdmin()
+
+const buildRequestCountDocuments = async () => {
+  const weeklyRequestCreation = {}
+
+  try {
+    const allRequests = admin
+      .firestore()
+      .collectionGroup('requests')
+
+    const snapshots = await allRequests.get()
+
+    await Promise.all(
+      snapshots.docs.map(async (doc) => {
+        console.log(`Adding request: ${doc.id} to the count`)
+
+        const data = doc.data()
+        const startDate = `${getFormattedStartOfWeek(firebaseTimestampToMoment(data.createdAt))}`
+        const endDate = `${getFormattedEndOfWeek(firebaseTimestampToMoment(data.createdAt))}`
+        const mapKeyForRequestDate = `${startDate}-${endDate}`
+
+        if (weeklyRequestCreation[mapKeyForRequestDate]) {
+          weeklyRequestCreation[mapKeyForRequestDate].amountOfRequests++
+        } else {
+          weeklyRequestCreation[mapKeyForRequestDate] = {
+            startDate,
+            endDate,
+            amountOfRequests: 1
+          }
+        }
+      })
+    )
+
+    Object.entries(weeklyRequestCreation).forEach(async ([key, data]) => {
+      await admin
+        .firestore()
+        .doc(`metric-amount-of-requests/${data.startDate}_${data.endDate}`)
+        .set(data, { merge: true })
+    })
+  } catch (error) {
+    console.error(error.message)
+  }
+}
+
+buildRequestCountDocuments()
+  .then()
```

Comment:
Empty `then` here?

Reply:
Yes, logging something here like 'Script ended' executes out of order and it would be weird to only have a `catch` block, as the general contract is to have a resolve/reject handlers even if they are empty.

Reply:
:thinking: how are you logging here? `then` receives a function. It can't be executed before the promise resolves:
```suggestion
  .then(() => console.log('Operation finished'))
```